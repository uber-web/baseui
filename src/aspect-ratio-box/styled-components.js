/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import {styled} from '../styles/index.js';

export const Root = styled<{$aspectRatio: number}>('div', ({$aspectRatio}) => ({
  width: '100%',
  position: 'relative',
  '::before': {
    content: '""',
    width: '1px',
    marginLeft: '-1px',
    float: 'left',
    height: 0,
    paddingTop: `${100 / $aspectRatio}%`,
    pointerEvents: 'none',
  },
  '::after': {
    content: '""',
    display: 'table',
    clear: 'both',
  },
}));

export const Body = styled<{}>('div', () => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  width: '100%',
}));
