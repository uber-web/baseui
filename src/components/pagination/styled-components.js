/* eslint-disable flowtype/require-valid-file-annotation */
import {styled} from '../../styles';

export const Root = styled('div', {
  display: 'flex',
});
