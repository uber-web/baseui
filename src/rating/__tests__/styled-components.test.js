/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {StyledRoot, StyledStar, StyledEmoticon} from '../styled-components';

describe('StyledComponents', () => {
  describe('StyledRoot', () => {
    it('displays correct styles by default', () => {
      const component = shallow(<StyledRoot />);
      expect(component.instance().getStyles()).toMatchSnapshot();
    });
  });

  describe('StyledStar', () => {
    it('displays correct styles by default', () => {
      const component = shallow(<StyledStar />);
      expect(component.instance().getStyles()).toMatchSnapshot();
    });

    it('displays correct styles if $isActive', () => {
      const component = shallow(<StyledStar $isActive />);
      expect(component.instance().getStyles()).toMatchSnapshot();
    });

    it('displays correct styles if $isSelected', () => {
      const component = shallow(<StyledStar $isSelected />);
      expect(component.instance().getStyles()).toMatchSnapshot();
    });
  });

  describe('StyledEmoticon', () => {
    it('displays correct styles by default', () => {
      const component = shallow(<StyledEmoticon $index={1} />);
      expect(component.instance().getStyles()).toMatchSnapshot();
    });

    it('displays correct styles if $isActive', () => {
      const component = shallow(<StyledEmoticon $isActive $index={1} />);
      expect(component.instance().getStyles()).toMatchSnapshot();
    });

    it('displays correct styles if $isSelected', () => {
      const component = shallow(<StyledEmoticon $isSelected $index={1} />);
      expect(component.instance().getStyles()).toMatchSnapshot();
    });

    describe('EmoticonStates', () => {
      it('displays correct SVG icon for each $index', () => {
        for (let x = 1; x <= 5; x++) {
          const component = shallow(<StyledEmoticon $index={x} />);
          expect(component.instance().getStyles()).toMatchSnapshot();
        }
      });
    });
  });
});
