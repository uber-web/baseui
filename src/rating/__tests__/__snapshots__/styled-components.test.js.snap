// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledComponents StyledEmoticon EmoticonStates displays correct SVG icon with $index=1 1`] = `
Object {
  ":after": Object {
    "content": "url('data:image/svg+xml;charset=utf8,%0A%20%20%20%20%3Csvg%20width%3D%2244%22%20height%3D%2244%22%20viewBox%3D%220%200%2044%2044%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%20%20%20%20%20%20%3Ccircle%20cx%3D%2222%22%20cy%3D%2222%22%20r%3D%2220%22%20fill%3D%22%24theme.colors.mono500%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M21.8377%2029C24.8073%2029%2026.6585%2030.7697%2027.1609%2031.5442C27.4615%2032.0075%2028.0807%2032.1395%2028.5441%2031.839C29.0074%2031.5384%2029.1394%2030.9192%2028.8388%2030.4558C28.0439%2029.2303%2025.614%2027%2021.8377%2027C18.0453%2027%2015.8091%2029.25%2015.1202%2030.5245C14.8576%2031.0103%2015.0385%2031.6171%2015.5244%2031.8797C16.0102%2032.1423%2016.617%2031.9614%2016.8796%2031.4755C17.2718%2030.75%2018.8842%2029%2021.8377%2029Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M11.684%2016.949C11.1601%2016.7743%2010.8769%2016.208%2011.0516%2015.684C11.2262%2015.1601%2011.7926%2014.8769%2012.3165%2015.0516L18.3165%2017.0516C18.8404%2017.2262%2019.1236%2017.7926%2018.949%2018.3165C18.8093%2018.7355%2018.4192%2019.0005%2018.0005%2019.0005C18.0004%2020.105%2017.105%2021.0002%2016.0005%2021.0002C14.896%2021.0002%2014.0005%2020.1048%2014.0005%2019.0002C14.0005%2018.5694%2014.1368%2018.1703%2014.3686%2017.8438L11.684%2016.949ZM32.949%2015.684C33.1237%2016.208%2032.8405%2016.7743%2032.3166%2016.949L29.6324%2017.8437C29.8642%2018.1702%2030.0005%2018.5693%2030.0005%2019.0002C30.0005%2020.1048%2029.1051%2021.0002%2028.0005%2021.0002C26.8961%2021.0002%2026.0007%2020.105%2026.0005%2019.0005C25.5817%2019.0007%2025.1914%2018.7356%2025.0516%2018.3165C24.877%2017.7926%2025.1602%2017.2262%2025.6841%2017.0516L31.6841%2015.0516C32.208%2014.8769%2032.7744%2015.1601%2032.949%2015.684Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%3C%2Fsvg%3E%0A%20%20')",
    "transition": "all $theme.animation.timing400",
  },
  ":focus": Object {
    "outline": "none",
  },
  "cursor": "pointer",
  "display": "inline-block",
  "height": "44px",
  "marginBottom": 0,
  "marginLeft": 0,
  "marginRight": "$theme.sizing.scale300",
  "marginTop": 0,
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
  "transform": "",
  "transition": "all $theme.animation.timing400",
  "width": "44px",
}
`;

exports[`StyledComponents StyledEmoticon EmoticonStates displays correct SVG icon with $index=2 1`] = `
Object {
  ":after": Object {
    "content": "url('data:image/svg+xml;charset=utf8,%0A%20%20%20%20%3Csvg%20width%3D%2244%22%20height%3D%2244%22%20viewBox%3D%220%200%2044%2044%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%20%20%20%20%20%20%3Ccircle%20cx%3D%2222%22%20cy%3D%2222%22%20r%3D%2220%22%20fill%3D%22%24theme.colors.mono500%22%2F%3E%0A%20%20%20%20%20%20%3Cellipse%20cx%3D%2216%22%20cy%3D%2218%22%20rx%3D%222%22%20ry%3D%222%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cellipse%20cx%3D%2228%22%20cy%3D%2218%22%20rx%3D%222%22%20ry%3D%222%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M21.8377%2028C24.8073%2028%2026.6585%2029.7697%2027.1609%2030.5442C27.4615%2031.0075%2028.0807%2031.1395%2028.5441%2030.839C29.0074%2030.5384%2029.1394%2029.9192%2028.8388%2029.4558C28.0439%2028.2303%2025.614%2026%2021.8377%2026C18.0453%2026%2015.8091%2028.25%2015.1202%2029.5245C14.8576%2030.0103%2015.0385%2030.6171%2015.5244%2030.8797C16.0102%2031.1423%2016.617%2030.9614%2016.8796%2030.4755C17.2718%2029.75%2018.8842%2028%2021.8377%2028Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%3C%2Fsvg%3E%0A%20%20')",
    "transition": "all $theme.animation.timing400",
  },
  ":focus": Object {
    "outline": "none",
  },
  "cursor": "pointer",
  "display": "inline-block",
  "height": "44px",
  "marginBottom": 0,
  "marginLeft": 0,
  "marginRight": "$theme.sizing.scale300",
  "marginTop": 0,
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
  "transform": "",
  "transition": "all $theme.animation.timing400",
  "width": "44px",
}
`;

exports[`StyledComponents StyledEmoticon EmoticonStates displays correct SVG icon with $index=3 1`] = `
Object {
  ":after": Object {
    "content": "url('data:image/svg+xml;charset=utf8,%0A%20%20%20%20%3Csvg%20width%3D%2244%22%20height%3D%2244%22%20viewBox%3D%220%200%2044%2044%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%20%20%20%20%20%20%3Ccircle%20cx%3D%2222%22%20cy%3D%2222%22%20r%3D%2220%22%20fill%3D%22%24theme.colors.mono500%22%2F%3E%0A%20%20%20%20%20%20%3Cellipse%20cx%3D%2216%22%20cy%3D%2219%22%20rx%3D%222%22%20ry%3D%222%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cellipse%20cx%3D%2228%22%20cy%3D%2219%22%20rx%3D%222%22%20ry%3D%222%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M15%2028C15%2027.4477%2015.4477%2027%2016%2027L16%2028L16%2029C15.4477%2029%2015%2028.5523%2015%2028ZM28%2028L28%2029H16L16%2028L16%2027H27.9995L28%2028ZM28%2028L28%2029C28.5523%2029%2029%2028.5523%2029%2028C29%2027.4477%2028.5518%2027%2027.9995%2027L28%2028Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%3C%2Fsvg%3E%0A%20%20')",
    "transition": "all $theme.animation.timing400",
  },
  ":focus": Object {
    "outline": "none",
  },
  "cursor": "pointer",
  "display": "inline-block",
  "height": "44px",
  "marginBottom": 0,
  "marginLeft": 0,
  "marginRight": "$theme.sizing.scale300",
  "marginTop": 0,
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
  "transform": "",
  "transition": "all $theme.animation.timing400",
  "width": "44px",
}
`;

exports[`StyledComponents StyledEmoticon EmoticonStates displays correct SVG icon with $index=4 1`] = `
Object {
  ":after": Object {
    "content": "url('data:image/svg+xml;charset=utf8,%0A%20%20%20%20%3Csvg%20width%3D%2244%22%20height%3D%2244%22%20viewBox%3D%220%200%2044%2044%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%20%20%20%20%20%20%3Ccircle%20cx%3D%2222%22%20cy%3D%2222%22%20r%3D%2220%22%20fill%3D%22%24theme.colors.mono500%22%2F%3E%0A%20%20%20%20%20%20%3Cellipse%20cx%3D%2216%22%20cy%3D%2218%22%20rx%3D%222%22%20ry%3D%222%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cellipse%20cx%3D%2228%22%20cy%3D%2218%22%20rx%3D%222%22%20ry%3D%222%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M21.8377%2029C24.8073%2029%2026.6585%2027.2303%2027.1609%2026.4558C27.4615%2025.9925%2028.0807%2025.8605%2028.5441%2026.161C29.0074%2026.4616%2029.1394%2027.0808%2028.8388%2027.5442C28.0439%2028.7697%2025.614%2031%2021.8377%2031C18.0453%2031%2015.8091%2028.75%2015.1202%2027.4755C14.8576%2026.9897%2015.0385%2026.3829%2015.5244%2026.1203C16.0102%2025.8577%2016.617%2026.0386%2016.8796%2026.5245C17.2718%2027.25%2018.8842%2029%2021.8377%2029Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%3C%2Fsvg%3E%0A%20%20')",
    "transition": "all $theme.animation.timing400",
  },
  ":focus": Object {
    "outline": "none",
  },
  "cursor": "pointer",
  "display": "inline-block",
  "height": "44px",
  "marginBottom": 0,
  "marginLeft": 0,
  "marginRight": "$theme.sizing.scale300",
  "marginTop": 0,
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
  "transform": "",
  "transition": "all $theme.animation.timing400",
  "width": "44px",
}
`;

exports[`StyledComponents StyledEmoticon EmoticonStates displays correct SVG icon with $index=5 1`] = `
Object {
  ":after": Object {
    "content": "url('data:image/svg+xml;charset=utf8,%0A%20%20%20%20%3Csvg%20width%3D%2244%22%20height%3D%2244%22%20viewBox%3D%220%200%2044%2044%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%20%20%20%20%20%20%3Ccircle%20cx%3D%2222%22%20cy%3D%2222%22%20r%3D%2220%22%20fill%3D%22%24theme.colors.mono500%22%2F%3E%0A%20%20%20%20%20%20%3Cellipse%20cx%3D%2216%22%20cy%3D%2218%22%20rx%3D%222%22%20ry%3D%222%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cellipse%20cx%3D%2228%22%20cy%3D%2218%22%20rx%3D%222%22%20ry%3D%222%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20d%3D%22M21.8378%2031C26.7027%2031%2028%2027.8%2028%2027H16C16%2027.8%2016.973%2031%2021.8378%2031Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M26.5528%2028H17.3513C17.4656%2028.1996%2017.6126%2028.3984%2017.8211%2028.6128C18.4608%2029.2891%2019.6453%2030%2021.8378%2030C24.0315%2030%2025.3168%2029.2869%2026.0488%2028.5805C26.2631%2028.3781%2026.4239%2028.1881%2026.5528%2028ZM27.4377%2030.0195C26.3048%2031.1131%2024.5091%2032%2021.8378%2032C19.1655%2032%2017.4311%2031.1109%2016.3681%2029.9872C15.3474%2028.9082%2015%2027.6646%2015%2027C15%2026.4477%2015.4477%2026%2016%2026H28C28.5523%2026%2029%2026.4477%2029%2027C29%2027.404%2028.8568%2027.9069%2028.6231%2028.3946C28.3936%2028.868%2027.9783%2029.4894%2027.4377%2030.0195Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%3C%2Fsvg%3E%0A%20%20')",
    "transition": "all $theme.animation.timing400",
  },
  ":focus": Object {
    "outline": "none",
  },
  "cursor": "pointer",
  "display": "inline-block",
  "height": "44px",
  "marginBottom": 0,
  "marginLeft": 0,
  "marginRight": "$theme.sizing.scale300",
  "marginTop": 0,
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
  "transform": "",
  "transition": "all $theme.animation.timing400",
  "width": "44px",
}
`;

exports[`StyledComponents StyledEmoticon displays correct styles by default 1`] = `
Object {
  ":after": Object {
    "content": "url('data:image/svg+xml;charset=utf8,%0A%20%20%20%20%3Csvg%20width%3D%2244%22%20height%3D%2244%22%20viewBox%3D%220%200%2044%2044%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%20%20%20%20%20%20%3Ccircle%20cx%3D%2222%22%20cy%3D%2222%22%20r%3D%2220%22%20fill%3D%22%24theme.colors.mono500%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M21.8377%2029C24.8073%2029%2026.6585%2030.7697%2027.1609%2031.5442C27.4615%2032.0075%2028.0807%2032.1395%2028.5441%2031.839C29.0074%2031.5384%2029.1394%2030.9192%2028.8388%2030.4558C28.0439%2029.2303%2025.614%2027%2021.8377%2027C18.0453%2027%2015.8091%2029.25%2015.1202%2030.5245C14.8576%2031.0103%2015.0385%2031.6171%2015.5244%2031.8797C16.0102%2032.1423%2016.617%2031.9614%2016.8796%2031.4755C17.2718%2030.75%2018.8842%2029%2021.8377%2029Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M11.684%2016.949C11.1601%2016.7743%2010.8769%2016.208%2011.0516%2015.684C11.2262%2015.1601%2011.7926%2014.8769%2012.3165%2015.0516L18.3165%2017.0516C18.8404%2017.2262%2019.1236%2017.7926%2018.949%2018.3165C18.8093%2018.7355%2018.4192%2019.0005%2018.0005%2019.0005C18.0004%2020.105%2017.105%2021.0002%2016.0005%2021.0002C14.896%2021.0002%2014.0005%2020.1048%2014.0005%2019.0002C14.0005%2018.5694%2014.1368%2018.1703%2014.3686%2017.8438L11.684%2016.949ZM32.949%2015.684C33.1237%2016.208%2032.8405%2016.7743%2032.3166%2016.949L29.6324%2017.8437C29.8642%2018.1702%2030.0005%2018.5693%2030.0005%2019.0002C30.0005%2020.1048%2029.1051%2021.0002%2028.0005%2021.0002C26.8961%2021.0002%2026.0007%2020.105%2026.0005%2019.0005C25.5817%2019.0007%2025.1914%2018.7356%2025.0516%2018.3165C24.877%2017.7926%2025.1602%2017.2262%2025.6841%2017.0516L31.6841%2015.0516C32.208%2014.8769%2032.7744%2015.1601%2032.949%2015.684Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%3C%2Fsvg%3E%0A%20%20')",
    "transition": "all $theme.animation.timing400",
  },
  ":focus": Object {
    "outline": "none",
  },
  "cursor": "pointer",
  "display": "inline-block",
  "height": "44px",
  "marginBottom": 0,
  "marginLeft": 0,
  "marginRight": "$theme.sizing.scale300",
  "marginTop": 0,
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
  "transform": "",
  "transition": "all $theme.animation.timing400",
  "width": "44px",
}
`;

exports[`StyledComponents StyledEmoticon displays correct styling with state $isActive 1`] = `
Object {
  ":after": Object {
    "content": "url('data:image/svg+xml;charset=utf8,%0A%20%20%20%20%3Csvg%20width%3D%2244%22%20height%3D%2244%22%20viewBox%3D%220%200%2044%2044%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%20%20%20%20%20%20%3Ccircle%20cx%3D%2222%22%20cy%3D%2222%22%20r%3D%2220%22%20fill%3D%22%24theme.colors.rating200%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M21.8377%2029C24.8073%2029%2026.6585%2030.7697%2027.1609%2031.5442C27.4615%2032.0075%2028.0807%2032.1395%2028.5441%2031.839C29.0074%2031.5384%2029.1394%2030.9192%2028.8388%2030.4558C28.0439%2029.2303%2025.614%2027%2021.8377%2027C18.0453%2027%2015.8091%2029.25%2015.1202%2030.5245C14.8576%2031.0103%2015.0385%2031.6171%2015.5244%2031.8797C16.0102%2032.1423%2016.617%2031.9614%2016.8796%2031.4755C17.2718%2030.75%2018.8842%2029%2021.8377%2029Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M11.684%2016.949C11.1601%2016.7743%2010.8769%2016.208%2011.0516%2015.684C11.2262%2015.1601%2011.7926%2014.8769%2012.3165%2015.0516L18.3165%2017.0516C18.8404%2017.2262%2019.1236%2017.7926%2018.949%2018.3165C18.8093%2018.7355%2018.4192%2019.0005%2018.0005%2019.0005C18.0004%2020.105%2017.105%2021.0002%2016.0005%2021.0002C14.896%2021.0002%2014.0005%2020.1048%2014.0005%2019.0002C14.0005%2018.5694%2014.1368%2018.1703%2014.3686%2017.8438L11.684%2016.949ZM32.949%2015.684C33.1237%2016.208%2032.8405%2016.7743%2032.3166%2016.949L29.6324%2017.8437C29.8642%2018.1702%2030.0005%2018.5693%2030.0005%2019.0002C30.0005%2020.1048%2029.1051%2021.0002%2028.0005%2021.0002C26.8961%2021.0002%2026.0007%2020.105%2026.0005%2019.0005C25.5817%2019.0007%2025.1914%2018.7356%2025.0516%2018.3165C24.877%2017.7926%2025.1602%2017.2262%2025.6841%2017.0516L31.6841%2015.0516C32.208%2014.8769%2032.7744%2015.1601%2032.949%2015.684Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%3C%2Fsvg%3E%0A%20%20')",
    "transition": "all $theme.animation.timing400",
  },
  ":focus": Object {
    "outline": "none",
  },
  "cursor": "pointer",
  "display": "inline-block",
  "height": "44px",
  "marginBottom": 0,
  "marginLeft": 0,
  "marginRight": "$theme.sizing.scale300",
  "marginTop": 0,
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
  "transform": "",
  "transition": "all $theme.animation.timing400",
  "width": "44px",
}
`;

exports[`StyledComponents StyledEmoticon displays correct styling with state $isSelected 1`] = `
Object {
  ":after": Object {
    "content": "url('data:image/svg+xml;charset=utf8,%0A%20%20%20%20%3Csvg%20width%3D%2244%22%20height%3D%2244%22%20viewBox%3D%220%200%2044%2044%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%20%20%20%20%20%20%3Ccircle%20cx%3D%2222%22%20cy%3D%2222%22%20r%3D%2220%22%20fill%3D%22%24theme.colors.mono500%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M21.8377%2029C24.8073%2029%2026.6585%2030.7697%2027.1609%2031.5442C27.4615%2032.0075%2028.0807%2032.1395%2028.5441%2031.839C29.0074%2031.5384%2029.1394%2030.9192%2028.8388%2030.4558C28.0439%2029.2303%2025.614%2027%2021.8377%2027C18.0453%2027%2015.8091%2029.25%2015.1202%2030.5245C14.8576%2031.0103%2015.0385%2031.6171%2015.5244%2031.8797C16.0102%2032.1423%2016.617%2031.9614%2016.8796%2031.4755C17.2718%2030.75%2018.8842%2029%2021.8377%2029Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%20%20%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M11.684%2016.949C11.1601%2016.7743%2010.8769%2016.208%2011.0516%2015.684C11.2262%2015.1601%2011.7926%2014.8769%2012.3165%2015.0516L18.3165%2017.0516C18.8404%2017.2262%2019.1236%2017.7926%2018.949%2018.3165C18.8093%2018.7355%2018.4192%2019.0005%2018.0005%2019.0005C18.0004%2020.105%2017.105%2021.0002%2016.0005%2021.0002C14.896%2021.0002%2014.0005%2020.1048%2014.0005%2019.0002C14.0005%2018.5694%2014.1368%2018.1703%2014.3686%2017.8438L11.684%2016.949ZM32.949%2015.684C33.1237%2016.208%2032.8405%2016.7743%2032.3166%2016.949L29.6324%2017.8437C29.8642%2018.1702%2030.0005%2018.5693%2030.0005%2019.0002C30.0005%2020.1048%2029.1051%2021.0002%2028.0005%2021.0002C26.8961%2021.0002%2026.0007%2020.105%2026.0005%2019.0005C25.5817%2019.0007%2025.1914%2018.7356%2025.0516%2018.3165C24.877%2017.7926%2025.1602%2017.2262%2025.6841%2017.0516L31.6841%2015.0516C32.208%2014.8769%2032.7744%2015.1601%2032.949%2015.684Z%22%20fill%3D%22black%22%2F%3E%0A%20%20%20%20%3C%2Fsvg%3E%0A%20%20')",
    "transition": "all $theme.animation.timing400",
  },
  ":focus": Object {
    "outline": "none",
  },
  "cursor": "pointer",
  "display": "inline-block",
  "height": "44px",
  "marginBottom": 0,
  "marginLeft": 0,
  "marginRight": "$theme.sizing.scale300",
  "marginTop": 0,
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
  "transform": "scale(1.1)",
  "transition": "all $theme.animation.timing400",
  "width": "44px",
}
`;

exports[`StyledComponents StyledRoot displays correct styles by default 1`] = `
Object {
  ":focus": Object {
    "outline": "none",
  },
  "display": "inline-block",
  "marginBottom": 0,
  "marginLeft": 0,
  "marginRight": 0,
  "marginTop": 0,
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
}
`;

exports[`StyledComponents StyledStar displays correct styles by default 1`] = `
Object {
  ":after": Object {
    "content": "url('data:image/svg+xml;charset=utf8,%0A%20%20%20%20%3Csvg%20width%3D%2222%22%20height%3D%2220%22%20viewBox%3D%220%200%2022%2020%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%20%20%20%20%20%20%3Cpath%0A%20%20%20%20%20%20%20%20d%3D%22M11%201.54816L13.1268%207.78053L13.2423%208.11905H13.6H20.4143L14.9132%2011.9714L14.6055%2012.1869L14.7268%2012.5424L16.8321%2018.7118L11.2868%2014.8285L11%2014.6277L10.7132%2014.8285L5.16792%2018.7118L7.27321%2012.5424L7.39454%2012.1869L7.08681%2011.9714L1.58566%208.11905H8.4H8.75769L8.87321%207.78053L11%201.54816Z%22%0A%20%20%20%20%20%20%20%20fill%3D%22%24theme.colors.mono300%22%0A%20%20%20%20%20%20%20%20stroke%3D%22%24theme.colors.mono500%22%0A%20%20%20%20%20%20%2F%3E%0A%20%20%20%20%3C%2Fsvg%3E%0A%20%20')",
    "transition": "all $theme.animation.timing400",
  },
  ":focus": Object {
    "outline": "none",
  },
  "cursor": "pointer",
  "display": "inline-block",
  "height": "20px",
  "marginBottom": 0,
  "marginLeft": 0,
  "marginRight": "$theme.sizing.scale300",
  "marginTop": 0,
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
  "transform": "",
  "transition": "all $theme.animation.timing400",
  "width": "22px",
}
`;

exports[`StyledComponents StyledStar displays correct styling with state $isActive 1`] = `
Object {
  ":after": Object {
    "content": "url('data:image/svg+xml;charset=utf8,%0A%20%20%20%20%3Csvg%20width%3D%2222%22%20height%3D%2220%22%20viewBox%3D%220%200%2022%2020%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%20%20%20%20%20%20%3Cpath%0A%20%20%20%20%20%20%20%20d%3D%22M11%201.54816L13.1268%207.78053L13.2423%208.11905H13.6H20.4143L14.9132%2011.9714L14.6055%2012.1869L14.7268%2012.5424L16.8321%2018.7118L11.2868%2014.8285L11%2014.6277L10.7132%2014.8285L5.16792%2018.7118L7.27321%2012.5424L7.39454%2012.1869L7.08681%2011.9714L1.58566%208.11905H8.4H8.75769L8.87321%207.78053L11%201.54816Z%22%0A%20%20%20%20%20%20%20%20fill%3D%22%24theme.colors.rating400%22%0A%20%20%20%20%20%20%20%20stroke%3D%22%24theme.colors.rating400%22%0A%20%20%20%20%20%20%2F%3E%0A%20%20%20%20%3C%2Fsvg%3E%0A%20%20')",
    "transition": "all $theme.animation.timing400",
  },
  ":focus": Object {
    "outline": "none",
  },
  "cursor": "pointer",
  "display": "inline-block",
  "height": "20px",
  "marginBottom": 0,
  "marginLeft": 0,
  "marginRight": "$theme.sizing.scale300",
  "marginTop": 0,
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
  "transform": "",
  "transition": "all $theme.animation.timing400",
  "width": "22px",
}
`;

exports[`StyledComponents StyledStar displays correct styling with state $isSelected 1`] = `
Object {
  ":after": Object {
    "content": "url('data:image/svg+xml;charset=utf8,%0A%20%20%20%20%3Csvg%20width%3D%2222%22%20height%3D%2220%22%20viewBox%3D%220%200%2022%2020%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%20%20%20%20%20%20%3Cpath%0A%20%20%20%20%20%20%20%20d%3D%22M11%201.54816L13.1268%207.78053L13.2423%208.11905H13.6H20.4143L14.9132%2011.9714L14.6055%2012.1869L14.7268%2012.5424L16.8321%2018.7118L11.2868%2014.8285L11%2014.6277L10.7132%2014.8285L5.16792%2018.7118L7.27321%2012.5424L7.39454%2012.1869L7.08681%2011.9714L1.58566%208.11905H8.4H8.75769L8.87321%207.78053L11%201.54816Z%22%0A%20%20%20%20%20%20%20%20fill%3D%22%24theme.colors.mono300%22%0A%20%20%20%20%20%20%20%20stroke%3D%22%24theme.colors.mono500%22%0A%20%20%20%20%20%20%2F%3E%0A%20%20%20%20%3C%2Fsvg%3E%0A%20%20')",
    "transition": "all $theme.animation.timing400",
  },
  ":focus": Object {
    "outline": "none",
  },
  "cursor": "pointer",
  "display": "inline-block",
  "height": "20px",
  "marginBottom": 0,
  "marginLeft": 0,
  "marginRight": "$theme.sizing.scale300",
  "marginTop": 0,
  "paddingBottom": 0,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 0,
  "transform": "scale(1.35)",
  "transition": "all $theme.animation.timing400",
  "width": "22px",
}
`;
